build_resolvers/lib/src/build_asset_uri_resolver.dart

```diff
@@ -130,16 +143,18 @@ class BuildAssetUriResolver extends UriResolver {
 
   @override
   Source resolveAbsolute(Uri uri, [Uri actualUri]) {
-    final cachedId = lookupAsset(uri);
-    if (cachedId == null) return null;
-    return resourceProvider
-        .getFile(assetPath(cachedId))
-        .createSource(cachedId.uri);
+    final assetId = parseAsset(uri);
+    if (assetId == null) {
+      return null;
+    }
+    final source =
+        resourceProvider.getFile(assetPath(assetId)).createSource(assetId.uri);
+    return source;
   }
 
   @override
   Uri restoreAbsolute(Source source) =>
-      lookupAsset(source.uri)?.uri ?? source.uri;
+      lookupCachedAsset(source.uri)?.uri ?? source.uri;
```

Comment:
I'm not sure if we should enforce the asset being cached here, or if we can just remove `lookupCachedAsset` entirely. The documentation from `UriResolver` says that `restoreAbsolute` "should be based solely on [source.fullName].", to me that sounds like further checks unnecessary. What do you think, @jakemac53?

Reply:
Tbh I would just go for the minimal change (what we have here). This code has always been very fragile and I am not sure what it might break.

@scheglov would know the most though.

Reply:
I am going to go ahead and merge/publish as this isn't changing the behavior. If we decide to change it further we can always do a followup :).